File: app/src/main/java/com/example/task3_database/MainActivity.java
package com.example.task3_database;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
private Button edit_categories_btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edit_categories_btn = findViewById(R.id.edit_categories);

        edit_categories_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent editCategoriesIntent = new Intent(MainActivity.this, CategoryMainModelActivity.class);
                startActivity(editCategoriesIntent);
            }
        });

    }
}

File: app/src/main/java/com/example/task4_vk/MainActivity.kt
package com.example.task4_vk

import android.os.Bundle
import android.content.Intent
import android.widget.EditText
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import android.widget.Toast


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val get_albums_btn: Button = findViewById(R.id.btnGetAlbums)
        val text_user_id: EditText = findViewById(R.id.inputUserIdText)
        get_albums_btn.setOnClickListener {
            val userId = text_user_id.text.toString()
            if (userId.isNotEmpty()) {
                val intent = Intent(this@MainActivity, AlbumsActivity::class.java)
                intent.putExtra("user_id", userId)
                startActivity(intent)
            }
            else{
                Toast.makeText(this, "Поле ID пропущено", Toast.LENGTH_SHORT).show()
            }
        }
    }
}